### PR TITLE
virt-handler: Replace CGO linking with a Go alternative

### PR DESCRIPTION
**Pull request by EdDev**
**What this PR does / why we need it**:

It is preferable to avoid CGO linking in virt-handler and limit it as much as possible to the virt-launcher (where it must link with the libvirt lib).

This is sometimes impossible, but in this case, the stable Linux API can be partially duplicated into the native Go code.

The CGO linkage had side effects on packages which imported the `virthandler` package, causing build issues.
(specifically, it collided with the e2e test binary)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

While building the e2e tests, the following error is seen on some platforms:
```
/lib64/libc.so.6: version `GLIBC_2.32'
/lib64/libc.so.6: version `GLIBC_2.34'
```

**Special notes for your reviewer**:
This PR attempts to resolve the e2e test build issues introduced by #8750 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by EdDev** [comment]
Storage [test failure](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8903/pull-kubevirt-e2e-k8s-1.24-sig-storage-root/1599488862600564736) has been addressed here: https://github.com/kubevirt/kubevirt/pull/8904

**Review by ormergi — APPROVED** [review]
Thanks for addressing this Edy.

**Comment by xpivarc** [comment]
/approve
/hold
@rmohr Looks good to you?

**Comment by jordigilh** [comment]
Can we test the real time lanes before merging?

**Comment by xpivarc** [comment]
> Can we test the real time lanes before merging?

Hi @jordigilh , the job succeeded.

**Comment by jordigilh** [comment]
> > Can we test the real time lanes before merging?
> 
> 
> 
> Hi @jordigilh , the job succeeded.

The real time lanes are not triggered by default. Since this PR addresses a change that impacts the code in these lanes, I think it is important to test them in this case.

**Comment by xpivarc** [comment]
> > > Can we test the real time lanes before merging?
> > 
> > 
> > Hi @jordigilh , the job succeeded.
> 
> The real time lanes are not triggered by default. Since this PR addresses a change that impacts the code in these lanes, I think it is important to test them in this case.

See:
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8903/pull-kubevirt-e2e-k8s-1.24-sig-compute-realtime-root/1599451708646232064
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8903/pull-kubevirt-e2e-k8s-1.24-sig-compute-realtime/1599451708012892160

**Comment by jordigilh** [comment]
> > > > Can we test the real time lanes before merging?
> 
> > > 
> 
> > > 
> 
> > > Hi @jordigilh , the job succeeded.
> 
> > 
> 
> > The real time lanes are not triggered by default. Since this PR addresses a change that impacts the code in these lanes, I think it is important to test them in this case.
> 
> 
> 
> See:
> 
> https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8903/pull-kubevirt-e2e-k8s-1.24-sig-compute-realtime-root/1599451708646232064
> 
> https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8903/pull-kubevirt-e2e-k8s-1.24-sig-compute-realtime/1599451708012892160

Terrific, I did not see them earlier.
/lgtm

**Pull request merged**
